pkg/storage/bloom/v1/bloom_tokenizer.go

```diff
@@ -216,10 +217,12 @@ outer:
 			for itr.Next() {
 				tok := itr.At()
 				tokens++
+
 				// TODO[owen-d]: [n]byte this
-				str := string(tok)
-				_, found := bt.cache[str] // A cache is used ahead of the SBF, as it cuts out the costly operations of scaling bloom filters
-				if found {
+				// To avoid allocations, an unsafe string can be used to check ownership in cache.
+				str := unsafe.String(unsafe.SliceData(tok), len(tok))
```

Comment:
note: The `[]byte` is reused across `Next` calls. I wonder if this impacts us here.
```
// The Token iterator uses shared buffers for performance. The []byte returned by At()
// is not safe for use after subsequent calls to Next()
```

Reply:
The unsafe string is only used for the membership check, not the assignment in the map.

Reply:
Oh good point. LGTM